cachelib/allocator/tests/BaseAllocatorTest.h

```diff
@@ -2469,6 +2470,85 @@ class BaseAllocatorTest : public AllocatorTest<AllocatorT> {
     }
   }
 
+  // tests case that correct parent item is acquired after move
+  void testChainedItemParentAcquireAfterMoveLoop() {
+
+    // create an allocator worth 250 slabs
+    // first slab is for overhead, second is parent class
+    // thrid is chained item 1 and rest are for new chained item alloc
+    // to move to.
+    std::unique_ptr<AllocatorT> alloc;
+    typename AllocatorT::Config config;
+    config.configureChainedItems();
+    config.setCacheSize(250 * Slab::kSize);
+
+    const std::set<uint32_t> allocSizes = {1024, 2048};
+    auto sizes = std::vector<uint32_t>{500, 1500};
+    std::atomic<uint64_t> numMoves{0};
+    std::atomic<uint64_t> numReplaces{0};
+    PoolId pid;
+
+    using Item = typename AllocatorT::Item;
+    config.enableMovingOnSlabRelease(
+        [&](Item& oldItem, Item& newItem, Item* parentPtr) {
+          assert(oldItem.getSize() == newItem.getSize());
+          assert(oldItem.isChainedItem());
+          std::memcpy(newItem.getMemory(), oldItem.getMemory(),
+                      oldItem.getSize());
+          folly::Latch latch(1);
+          auto insertThread = std::make_unique<std::thread>([&](){
+            ASSERT_NO_THROW({
+              auto parentReplacement = alloc->allocate(pid,parentPtr->getKey(),sizes[0]);
+              Item *parentCopy = parentPtr;
+              latch.count_down();
+              while (parentCopy->isMoving());
+              alloc->insertOrReplace(parentReplacement);
+              ++numReplaces;
+            });
+          });
+          insertThread->detach();
+          latch.wait();
+          ++numMoves;
+        }
+    );
+
+    alloc = std::make_unique<AllocatorT>(config);
+
+    const size_t numBytes = alloc->getCacheMemoryStats().ramCacheSize;
+    const auto poolSize = numBytes;
+    pid = alloc->addPool("one", poolSize, allocSizes);
+
+    auto allocFn = [&](std::string keyPrefix, std::vector<uint32_t> sizes) {
+      for (unsigned int loop = 0; loop < 20; ++loop) {
+        std::vector<typename AllocatorT::WriteHandle> parentHandles;
```

Comment:
`parentHandles` is not used?

Reply:
removed - it was leftover from initial test that I used as base